### ISSUE TITLE
Add Azure pipelines support (global tool)

### ISSUE THREAD

**Issue by srollinet** [cake-build/cake#2357]
### What You Are Seeing?

Impossible to install cake as a global tool and use it in azure pipelines, using a linux agent

### What is Expected?

It should be possible to install cake (if possible as a global tool) and use it to build on a linux agent.

### What version of Cake are you using?

0.30.0

### Are you running on a 32 or 64 bit system?

64

### What environment are you running on?  Windows? Linux? Mac?

Linux (Hosted Ubuntu 1604)

### Are you running on a CI Server?  If so, which one?

Azure pipelines

#### attempt 1
Tried the predefined "Cake" task in Azure : failed because it only support Windows agents

#### attempt 2
Install cake as global tool using ".NET Core Global Tool Installer" task
Tool name: `cake.tool`
Tool version: `0.30.0`
result: 
```
2018-11-14T09:11:03.4765167Z [command]/usr/bin/dotnet tool install --tool-path /home/vsts/work/_temp/90c078c5-8d17-4dfa-8177-908a29b7ff0d Cake.Tool --version 0.30.0
2018-11-14T09:11:27.6021255Z You can invoke the tool using the following command: dotnet-cake
2018-11-14T09:11:27.6022290Z Tool 'cake.tool' (version '0.30.0') was successfully installed.
2018-11-14T09:11:27.6069000Z Error: could not detect installed version
```

#### attempt 3
install cake and build using a "Command line" task
command:
```
dotnet tool install --global Cake.Tool --version 0.30.0
dotnet cake //also tried with dotnet-cake
```

result:
```
2018-11-14T09:21:49.9480514Z [command]/bin/bash --noprofile --norc /home/vsts/work/_temp/3b25e7be-3635-41e1-9ac7-3e208c4f9533.sh
2018-11-14T09:22:16.9372187Z Since you just installed the .NET Core SDK, you will need to logout or restart your session before running the tool you installed.
2018-11-14T09:22:16.9373384Z You can invoke the tool using the following command: dotnet-cake
2018-11-14T09:22:16.9373698Z Tool 'cake.tool' (version '0.30.0') was successfully installed.
2018-11-14T09:22:16.9418563Z /home/vsts/work/_temp/3b25e7be-3635-41e1-9ac7-3e208c4f9533.sh: line 2: dotnet-cake: command not found
2018-11-14T09:22:16.9527061Z ##[error]Bash exited with code '127'.
```

Maybe there is another way I haven't tried?

**Comment by devlead** [comment]
@srollinet It's likely not found because the .NET tool folder isn't in path of the profile you're running under.

You would need to add something like `export PATH="$PATH:/home/{USERNAME}/.dotnet/tools"` to your bash profile (`~/.bash_profile`) to just be able to execute `dotnet cake`.

That said what I usually do is by task looks something like this

```yaml
- bash: |
   export DOTNET_SKIP_FIRST_TIME_EXPERIENCE=1
   export DOTNET_CLI_TELEMETRY_OPTOUT=1
   
   dotnet tool install -g --version 0.30.0 Cake.Tool
   
   ~/.dotnet/tools/dotnet-cake 
```

Another option is to install it in the current folder
```bash
dotnet tool install --tool-path ./tools --version 0.30.0 Cake.Tool
./tools/dotnet-cake
```

Advantage with doing it in repo folder is that you can have different versions of Cake pinned for different projects, which ensures more reproducible builds. An example script that does this and also handles the bootstrapping of module directives.
https://github.com/csharpfritz/CoreWiki/blob/dev/build.sh

**Comment by HofmeisterAn** [comment]
I use your second option. However, it’s a bit inconvenient with templates and different vmImages. For example Linux and macOS uses `./tools/dotnet-cake`, Windows on the other hand `.\tools\dotnet-cake`. I use parameters to pass a different style to the template job, is there any better approach?

**Comment by srollinet** [comment]
The issue with global tool installer seem to be solved. So I guess the problem was with the global tool installer.

@HofmeisterAn So now, you can install cake using the step `.NET Core Global Tool Installer` and run it using a step `.NET Core` with custom command `cake`

**Comment by HofmeisterAn** [comment]
With `./tools/dotnet-cake` I get following error:

```
2018-11-27T06:39:10.0645961Z ./tools/dotnet-cake --target=Build
2018-11-27T06:39:10.1222880Z ##[command]"C:\Windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\b6f2b29a-2f62-4e80-a77b-18ced2ee9b12.cmd""
2018-11-27T06:39:10.1317912Z '.' is not recognized as an internal or external command,
2018-11-27T06:39:10.1318243Z operable program or batch file.
2018-11-27T06:39:10.2335475Z ##[error]Cmd.exe exited with code '1'.
```

Even without `./` the build will fail on Windows images.

```
2018-11-27T06:46:18.5902331Z tools/dotnet-cake --target=Build
2018-11-27T06:46:18.6444832Z ##[command]"C:\Windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\6a8236da-4a86-4085-a0a2-ed2bc5b4c14c.cmd""
2018-11-27T06:46:18.6551754Z 'tools' is not recognized as an internal or external command,
2018-11-27T06:46:18.6551903Z operable program or batch file.
2018-11-27T06:39:10.2335475Z ##[error]Cmd.exe exited with code '1'.
```

I didn't invest much time yesterday, I just had a quick look. I noticed that `\` will work.

@srollinet I will check that out later, thanks!

**Comment by devlead** [comment]
@HofmeisterAn ah that would because cmd.exe, PowerShell task is now xplat and tried this on both posix and Windows
```yaml
steps:
- powershell: |
   dotnet tool install --tool-path ./tools --version 0.30.0 Cake.Tool
   ./tools/dotnet-cake --version
```

@srollinet excellent, yes there were an issue with the .NET CLI on Azure DevOps belive they fixed it in 2.1.500


You can ensure and pin which version of .NET Core you're executing with the .NET Core installer
```yaml
- task: DotNetCoreInstaller@0
  inputs:
    packageType: 'sdk'
    version: '2.1.500' 
```
